src/web/app/components/question-responses/single-statistics/single-statistics.component.spec.ts

```diff
@@ -21,7 +26,55 @@ describe('SingleStatisticsComponent', () => {
     fixture.detectChanges();
   });
 
+  /**
+   * Mock component responses list for all scenarios.
+   *
+   * @param isMissingResponse - is for setting true/false in the isMissingResponse field if missing response.
+   * @param questionType - is for updating feedback question type field
+   * @param recipient - is for updating recipient field
+   */
+  function getResponseOutput(isMissingResponse: boolean,
+                             questionType: FeedbackQuestionType,
+                             recipient: string = 'recipient'): ResponseOutput[] {
+    return [
+      {
+        isMissingResponse,
+        responseId: 'some_id',
+        giver: 'some_giver',
+        giverTeam: 'some_team',
+        giverSection: 'section_1',
+        recipient,
+        recipientTeam: 'recipient_1_team',
+        recipientSection: 'recipient_1_section',
+        responseDetails: {
+          questionType,
+        },
+        instructorComments: [],
+      },
+    ];
+  }
+
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('should return empty responsesToUse when missing response & feedback question type is not contrib', () => {
+    component.responses = getResponseOutput(true, FeedbackQuestionType.TEXT);
+    fixture.detectChanges();
+    expect(component.responsesToUse.length).toBe(0);
+  });
+
+  it('should return empty when is using response to self & response recipient is not You', () => {
+    component.isStudent = true;
+    component.responses = getResponseOutput(false, FeedbackQuestionType.NUMSCALE);
+    fixture.detectChanges();
+    expect(component.responsesToUse.length).toBe(0);
+  });
+
+  it('should return responses based on selected section when ngOnInit is called', () => {
```

Comment:
this description is not accurate, as we do not return responses based on selected section, rather we delegate the filtering to isFeedbackResponsesDisplayedOnSection, if none of the above conditions (missing responses, question type is not contrib, isUsingResponseToSelf and response recipient is not You) are true

Reply:
hi @sarfophil do change the description of this before we merge this PR!